Mathlib/RingTheory/Flat/Basic.lean

```diff
@@ -229,6 +240,146 @@ instance of_projective [h : Projective R M] : Flat R M := by
   cases h with
     | _ e he => exact of_retract R _ _ _ _ he
 
+open BigOperators in
+/--
+Define character module of `M` to be `M → ℚ ⧸ ℤ`
+If the character module of `M` is an injective module then `L ⊗ 𝟙 M` is injective for any linear map
+`L`.
+-/
+lemma rTensor_preserves_injective_linearMap_of_injective_characterModule
+    (h : Module.Injective R <| CharacterModule M) :
+    ∀ ⦃N N' : Type v⦄ [AddCommGroup N] [AddCommGroup N'] [Module R N] [Module R N']
+      (L : N →ₗ[R] N'), Function.Injective L → Function.Injective (L.rTensor M) := by
+  intros A B _ _ _ _ L hL
+  rw [← LinearMap.ker_eq_bot, eq_bot_iff]
+  rintro z (hz : _ = 0)
+  -- Consider an injective linear map `L : A → B`, we want to prove that `(L ⊗ 𝟙 M) z = 0`
+  -- implies `z = 0`
```

Comment:
```suggestion
  -- Consider an injective linear map `L : A → B`, we want to prove that
  -- `(L ⊗ 𝟙 M) z = 0` implies `z = 0`
```
I was very confused until I noticed that the line break wasn't the end of the sentence :-)

Reply:
Junyan golfed all this away!